docs/tutorial/custom_frontend.rst

```diff
@@ -0,0 +1,323 @@
+directory and a ``snap`` directory::
+
+  checkbox_tutorial:
+  ├─ providers
+  │  ├── tutorial
+  │  │   ├─── manage.py
+  │  [...]
+  └── snap
+     └── snapcraft.yaml
+
+In this tutorial we will package only one provider, the tutorial provider, but
+this structure is future-proof, allowing you to expand your custom frontend
+snap as needed.
+
+.. warning::
+
+   This tutorial will focus on creating a core24 snap. If you need a snap for
+   another base (remember that the base should match your system) you will need
+   to adjust the base of the snap and the runtime version you pull. They must
+   match (base: core24 -> checkbox24). If they don't match, you may encounter
+   several very hard to debug issues so keep this in mind!
+
+Edit the ``snapcraft.yaml`` and set it to the following:
+
+.. code:: yaml
+
+  name: checkbox-tutorial
+  summary: Checkbox tutorial custom frontend
+  description: |
+    This is a custom frontend for Checkbox containing the tutorial provider.
+    Use it along the checkbox24 snap.
+  grade: stable
+  confinement: strict
+  base: core24
+
+  adopt-info: version-calculator
+
+  slots:
+    custom-frontend:
+      interface: content
+      content: custom-frontend
+      read:
+        - /
+
+  package-repositories:
+    - type: apt
+      ppa: checkbox-dev/edge # used to pull the installation machinery
+
+  parts:
+    version-calculator:
+      plugin: dump
+      source: snap
+      override-pull: |
+        craftctl default
+        # consider calculating the version here!
+        export version="v1.2.3"
+        [ $version ] || exit 1
+        craftctl set version=$version
+    checkbox-provider-tutorial:
+      # install each provider in its own independent part
+      plugin: dump
+      source: providers/tutorial
+      source-type: local
+      build-environment:
+        - PYTHONPYCACHEPREFIX: "/tmp"
+      build-packages: # add here any other build dependency you may have
+        - checkbox-ng # this is necessary to run mange.py
+      stage-packages: # add here any runtime dependency of your provider
+        - jq          # Note: we added a packaging metadata unit but those are
+                      #       not used in snap builds! Repeat all dependencies
+                      #       here if you want them included in the snap
+      override-build: |
+        # Do NOT validate your provider, if you depend on tests/nested parts
+        # that aren't in this provider (or in this repository) it will fail to
+        # validate. We will handle validation in a separate workflow.
+        python3 manage.py build
+        # Note: providers MUST be in /providers to use the new custom-frontend
+        #       connection, do not change the location
+        python3 manage.py install --layout=relocatable --prefix=/providers/tutorial --root="$CRAFT_PART_INSTALL"
+
+Let's try to build our new frontend snap:
+
+.. code:: shell-session
+
+   $ sudo snap install --classic snapcraft
```

Comment:
Snapcraft should be prepared before starting this tutorial, so I'd suggest moving this command into a small "Prerequisite" section at the very beginning